src/boot/efi/fuzz-bcd.c

```diff
@@ -0,0 +1,37 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "alloc-util.h"
+#include "fd-util.h"
+#include "fuzz.h"
+#include "utf8.h"
+
+#include "bcd.c"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
```

Comment:
You're overcomplicating this way too much. I've been running witht his:
```c
/* SPDX-License-Identifier: LGPL-2.1-or-later */

#include "alloc-util.h"
#include "fuzz.h"
#include "log.h"
#include "utf8.h"

#include "bcd.c"

int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
        _cleanup_free_ uint8_t *copy = memdup(data, size);
        assert_se(copy);

        char16_t *title = get_bcd_title(copy, size);
        assert_se(!title || char16_strlen(title) >= 0);

        return 0;
}
```

Reply:
If it's safe to assume that the parser will never log anything or trigger OOMs by reading data without any limits I agree it can be trimmed significantly. But I generally don't try to make any assumptions about code :-)

Reply:
I obviously agree that the if expression where `/dev/null` is opened can be safely replaced with
```
assert_se(!title || char16_strlen(title) >= 0);
```
though :-)

Reply:
The bcd parser neither logs nor allocates memory.

Reply:
If it receives 4Gb files can it always finish in less than 10 seconds?

> The bcd parser neither logs nor allocates memory.

Currently it doesn't. I have no idea what is going to happen to it in 2 years though.

Reply:
If you really wanna restrict the size, shouldn't you be telling the fuzzer directly like with fuzz-env-file?

Reply:
That ".options" file is compatible with libFuzzer only. It has been kind of broken since AFL and honggfuzz started to build systemd on OSS-Fuzz. And as far as I can see to address that `fuzz-env-file` limits its input explicitly by calling
```
        if (size == 0 || size > 65535)
                return 0;
```

Reply:
I replaced that if expression with `assert_se(!title || char16_strlen(title) >= 0);` and decided to keep the other code mostly because I don't want to receive bogus notifications from OSS-Fuzz about pointless OOMs, timeouts and so on in the foreseeable future.